include/aspect/geometry_model/spherical_shell.h

```diff
@@ -95,6 +95,18 @@ namespace aspect
 
           /**
            * @copydoc Manifold::normal_vector()
+           *
+           * We fudge here, but for a good reason. What the function is supposed
+           * to compute is the normal vector to the surface. This *should* be the
+           * normal to the surface with topography, but instead we return the
+           * normal to the undeformed surface -- i.e., the radial direction. This
+           * is, in particular, used to compute no-flux boundary conditions,
```

Comment:
My understanding is that VectorTools::compute_no_normal_flux_constraints() ignores the Manifold normal while VectorTools::compute_no_normal_flux_constraints_on_level() uses it. I should document this inside deal.II...

Reply:
Yes please!